src/EnergyPlus/WaterCoils.cc

```diff
@@ -2622,8 +2638,19 @@ namespace WaterCoils {
                     WaterCoil(CoilNum).DesInletAirHumRat = TempSize; // coil report
                     WaterCoil(CoilNum).InletAirHumRat = TempSize;
                     TempSize = AutoSize; // these data are initially 0, set to autosize to receive a result from RequestSizing
-                    RequestSizing(state, CompType, CompName, HeatingAirflowUASizing, SizingString, TempSize, bPRINT, RoutineName);
+                    //TODO: RequestSizing(state, CompType, CompName, HeatingAirflowUASizing, SizingString, TempSize, bPRINT, RoutineName);
                     // WaterCoil( CoilNum ).InletAirMassFlowRate = TempSize;
+                    EnergyPlus::HeatingAirflowUASizer sizer;
+                    EnergyPlus::HeatingAirflowUASizerFlags flags;
+                    flags.curTermUnitSizingNum = DataSizing::CurTermUnitSizingNum;
+                    flags.curZoneEqNum = DataSizing::CurZoneEqNum;
+                    sizer.zoneSizingInput = DataSizing::ZoneSizingInput;
+                    flags.termUnitSingDuct = TermUnitSingDuct;
+                    flags.termUnitPIU = TermUnitPIU;
+                    flags.termUnitIU = TermUnitIU;
+                    sizer.setParameters(baseFlags, flags, DataSizing::TermUnitSizing, DataSizing::FinalZoneSizing, DataSizing::ZoneEqSizing);
+                    AutoSizingResultType result = sizer.size(TempSize);
+                    WaterCoil(CoilNum).InletAirMassFlowRate = sizer.autoSizedValue;
```

Comment:
Any reason why we couldn't move the setParameters call inside the size function? And reduce this to a 1-line call? Seems all the flags could be set up in the function except for maybe the print flag ... which could be an argument.

Reply:
Also the bool result from the size function. All sizing inputs are reals so maybe change that and pass an errorsFound flag? And could this be a 1-liner?

EnergyPlus::HeatingAirflowUASizer sizer;
WaterCoil(CoilNum).InletAirMassFlowRate = sizer.size(TempSize, errorsFound)

Reply:
WaterCoil(CoilNum).InletAirMassFlowRate = EnergyPlus::HeatingAirflowUASizer.size(TempSize, errorsFound)